### ISSUE TITLE
[function-paren-newline] false positives with parentheses in return types

### ISSUE THREAD

**Issue by kku39** [typescript-eslint/typescript-eslint#3318]
**Repro**

```JSON
{
  "rules": {
    "function-paren-newline": ["error", "consistent"]
  }
}
```

```TS
const foo = (
): (() => void) => {
  return () => {};
};
```

or

```TS
const foo = (
): { onCall: () => void } => {
  return { onCall: () => {} };
};
```

**Expected Result**

No error.

**Actual Result**

> 2:15   error  Expected newline before ')'               function-paren-newline

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.0` |
| `@typescript-eslint/parser`        | `4.22.0` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.25.0` |
| `node`                             | `14.15.4` |

**Comment by tanohzana** [comment]
I am working on it!

**Comment by shpakkdv** [comment]
@tanohzana any updates?

just to link: https://github.com/typescript-eslint/typescript-eslint/issues/942

**Comment by tanohzana** [comment]
@shpakkdv  I haven't had that much time to work on this. Let's see if I can do it before the end of the year 👍🏽

**Comment by shpakkdv** [comment]
@tanohzana  Sounds great! I feel a strong lack of this rule in TS 😅
Thank you! 👍

**Comment by shpakkdv** [comment]
@tanohzana are there any changes on this?

**Comment by Josh-Cena** [comment]
Because the ESLint core is also working on [deprecating/removing its formatting rules](https://github.com/eslint/eslint/issues/17522), we will close this. A 3rd-party plugin providing formatting, like https://github.com/eslint-stylistic/eslint-stylistic, can surely support TS OOTB.
